src/video.ts

```diff
@@ -1,58 +1,75 @@
+    [ 'minduration', isInteger ],
+    [ 'maxduration', isInteger ],
+    [ 'startdelay', isInteger ],
+    [ 'maxseq', isInteger ],
+    [ 'poddur', isInteger ],
+    [ 'protocols', isArrayOfNums ],
+    [ 'w', isInteger ],
+    [ 'h', isInteger ],
+    [ 'podid', isStr ],
+    [ 'podseq', isInteger ],
+    [ 'rqddurs', isArrayOfNums ],
+    [ 'placement', isInteger ], // deprecated, see plcmt
+    [ 'plcmt', isInteger ],
+    [ 'linearity', isInteger ],
+    [ 'skip', value => [1, 0].includes(value) ],
+    [ 'skipmin', isInteger ],
+    [ 'skipafter', isInteger ],
+    [ 'sequence', isInteger ], // deprecated
+    [ 'slotinpod', isInteger ],
+    [ 'mincpmpersec', isNumber ],
+    [ 'battr', isArrayOfNums ],
+    [ 'maxextended', isInteger ],
+    [ 'minbitrate', isInteger ],
+    [ 'maxbitrate', isInteger ],
+    [ 'boxingallowed', isInteger ],
+    [ 'playbackmethod', isArrayOfNums ],
+    [ 'playbackend', isInteger ],
+    [ 'delivery', isArrayOfNums ],
+    [ 'pos', isInteger ],
+    [ 'api', isArrayOfNums ],
+    [ 'companiontype', isArrayOfNums ],
+    [ 'poddedupe', isArrayOfNums ]
+] as const;
+
 /**
  * List of OpenRTB 2.x video object properties with simple validators.
  * Not included: `companionad`, `durfloors`, `ext`
  * reference: https://github.com/InteractiveAdvertisingBureau/openrtb2.x/blob/main/2.6.md
  */
-export const ORTB_VIDEO_PARAMS = new Map([
-  [ 'mimes', value => Array.isArray(value) && value.length > 0 && value.every(v => typeof v === 'string') ],
-  [ 'minduration', isInteger ],
-  [ 'maxduration', isInteger ],
-  [ 'startdelay', isInteger ],
-  [ 'maxseq', isInteger ],
-  [ 'poddur', isInteger ],
-  [ 'protocols', isArrayOfNums ],
-  [ 'w', isInteger ],
-  [ 'h', isInteger ],
-  [ 'podid', isStr ],
-  [ 'podseq', isInteger ],
-  [ 'rqddurs', isArrayOfNums ],
-  [ 'placement', isInteger ], // deprecated, see plcmt
-  [ 'plcmt', isInteger ],
-  [ 'linearity', isInteger ],
-  [ 'skip', value => [1, 0].includes(value) ],
-  [ 'skipmin', isInteger ],
-  [ 'skipafter', isInteger ],
-  [ 'sequence', isInteger ], // deprecated
-  [ 'slotinpod', isInteger ],
-  [ 'mincpmpersec', isNumber ],
-  [ 'battr', isArrayOfNums ],
-  [ 'maxextended', isInteger ],
-  [ 'minbitrate', isInteger ],
-  [ 'maxbitrate', isInteger ],
-  [ 'boxingallowed', isInteger ],
-  [ 'playbackmethod', isArrayOfNums ],
-  [ 'playbackend', isInteger ],
-  [ 'delivery', isArrayOfNums ],
-  [ 'pos', isInteger ],
-  [ 'api', isArrayOfNums ],
-  [ 'companiontype', isArrayOfNums ],
-  [ 'poddedupe', isArrayOfNums ]
-]);
-
-export function fillVideoDefaults(adUnit) {
+export const ORTB_VIDEO_PARAMS = new Map(ORTB_PARAMS);
+
+export type VideoContext = typeof INSTREAM | typeof OUTSTREAM | typeof ADPOD;
+
+export interface VideoMediaType extends BaseMediaType, Pick<ORTBImp['video'], (typeof ORTB_PARAMS)[number][0]> {
+    context: VideoContext;
+    /**
+     * @deprecated - use w & h instead.
```

Comment:
OH DAMN! This went entirely passed me.

Reply:
I jumped the gun on this - it made sense to me as w/h are better (it's ortb, and not all adapters support multiple sizes) but we should discuss this with the video committee. for now I added some logic to sync playerSize and w/h and removed the deprecation label.